rosidl_parser/rosidl_parser/__init__.py

```diff
@@ -705,3 +709,75 @@ def parse_service_string(pkg_name, srv_name, message_string):
         pkg_name, srv_name + SERVICE_RESPONSE_MESSAGE_SUFFIX, response_message_string)
 
     return ServiceSpecification(pkg_name, srv_name, request_message, response_message)
+
+def parse_action_file(pkg_name, interface_filename):
+    basename = os.path.basename(interface_filename)
+    action_name = os.path.splitext(basename)[0]
+    with open(interface_filename, 'r') as h:
+        return parse_action_string(pkg_name, action_name, h.read())
+
+
+def parse_action_string(pkg_name, action_name, message_string):
+    lines = message_string.splitlines()
+    separator_indices = [
+        index for index, line in enumerate(lines) if line == SERVICE_REQUEST_RESPONSE_SEPARATOR]
+    if not separator_indices:
+        raise InvalidServiceSpecification(
+            "Could not find separator '%s' between action goal, result and feedback services" %
+            SERVICE_REQUEST_RESPONSE_SEPARATOR)
+    if len(separator_indices) != 3:
+        raise InvalidServiceSpecification(
+            "Could not find unique separator '%s' between action goal, result and \
+            feedback services" %
+            SERVICE_REQUEST_RESPONSE_SEPARATOR)
```

Comment:
@sservulo nit: we could also check directly for the amount of separators (or sections) and complain about bad action definition. Also, I'd wouldn't phrase exception messages in terms of services, it's somewhat confusing.